python/dgl/dataloading/negative_sampler.py

```diff
@@ -65,3 +66,67 @@ def _generate(self, g, eids, canonical_etype):
         src = F.repeat(src, self.k, 0)
         dst = F.randint(shape, dtype, ctx, 0, g.number_of_nodes(vtype))
         return src, dst
+
+# Alias
+Uniform = PerSourceUniform
+
+class GlobalUniform(_BaseNegativeSampler):
+    """Negative sampler that randomly chooses negative source-destination pairs according
+    to a uniform distribution.
+
+    For each edge ``(u, v)`` of type ``(srctype, etype, dsttype)``, DGL generates
```

Comment:
I'm confused by this.  This almost sounds like `k=2` means you'll get `2*g.num_edges()` returned, not just 2.  Maybe we can start this doc string at "DGL generates..." and exclude this first phrase?  I'm also confused by how srctype, etype and dsttype come into play.  Which parts are specified by the user, what assumptions are made, and how does it impact the results?

Reply:
Since the positive examples can have different edge types (i.e. the `eids` will be a dictionary), I'm trying to say DGL will generate k negative pairs per edge with the same edge type.

Reply:
Because the number of edges per type could vary across minibatches, I think it's not a good idea to fix the number of total negative samples to generate per minibatch.  I also changed the implementation to generate `len(eid) * k` negative samples.